### PR TITLE
Automate Kolibri Windows Installer signing

### PR DESCRIPTION
**Pull request by mrpau-richard**
### Summary
I created a script to automate Kolibri windows installer signing at buildkite.

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by mrpau-richard** [comment]
To test this set the `SIGN_WINDOWS_INSTALLER=true` before running the build at the buidkite.
Here's the sample screenshot.
screen shot 2018-05-28 at 5 04 20 pm

Here's the build output.
https://buildkite.com/learningequality/mrpau-windows-sign-testing/builds/54

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -21,3 +21,8 @@ steps:
 
   - label: Upload artifacts
     command: .buildkite/setup_and_upload_artifact.sh && docker image prune -f -a
+
+  - label: Sign Windows installer
+    command: .buildkite/sign_windows_installer.bat
+    agents:
+      queue: "windows-sign"
```
Comment (aronasorman) [inline .buildkite/pipeline.yml:28]: Does this mean only agents with this tag is able to sign this? So vader won't start signing?

**Review by aronasorman — COMMENTED** [review]
Have you tested if the produced signed installer works on windows and doesn't prompt with the 'unknown developer' dialog box?

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (radinamatic) [reply .buildkite/pipeline.yml:28]: I wasn't cc-ed on this PR, but will test asap!

Comment (radinamatic) [reply .buildkite/pipeline.yml:28]: It says that the build failed, but the artifact is there... :thinking:

Comment (radinamatic) [reply .buildkite/pipeline.yml:28]: This one does not seem signed, running a new build just in case...

selection_168

**Comment by radinamatic** [comment]
Ok, so not even after starting the new build and adding the env var, the Windows artifact is not signed... :confused: 

selection_169

Build fails at the last step (upload), so that can't be the issue, right?

**Comment by indirectlylit** [comment]
@radinamatic my guess is that some initial configuration of the buildkite environment is required for the signing to take effect - is that right @mrpau-richard?

**Comment by radinamatic** [comment]
@indirectlylit Fair enough!
@mrpau-richard Could you add that to the instructions on how to properly test the build (I followed those in the screenshot), and I'll re-run it!

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (mrpau-richard) [reply .buildkite/pipeline.yml:28]: >Have you tested if the produced signed installer works on windows and doesn't prompt with the 'unknown developer' dialog box?

Yeah I tested it, here's the sample build [artifact link](https://buildkite.com/organizations/learningequality/pipelines/mrpau-windows-sign-testing/builds/54/jobs/6f41ee33-996f-4ccc-9900-29eeb4e23bf2/artifacts/790970f3-768f-4650-99d8-bb8c0e08b847) .

>Does this mean only agents with this tag is able to sign this? So vader won't start signing?

The Windows signing credentials is on the `Windows-2016` agent. If we sign it at vader we need to add those signing credentials there.

**Comment by mrpau-richard** [comment]
@radinamatic sorry I forgot to change the branch at my screenshot. My changes is not in develop yet. 
You should follow this screenshot build.
screen shot 2018-05-30 at 9 08 59 am

And build it on this [buildkite pipeline](https://buildkite.com/learningequality/mrpau-windows-sign-testing)

The signed installer artifact will be uploaded at `Sign Windows installer` step.
screen shot 2018-05-30 at 9 13 28 am

If the environment variable is not set I will show an error log (refer to the screenshot below). and it will not upload the installer artifact.
screen shot 2018-05-30 at 9 16 37 am

@radinamatic  after this PR merge you can build it at [kolibri pipeline](https://buildkite.com/learningequality/kolibri/) and set the branch on develop.

**Comment by aronasorman** [comment]
@mrpau-richard can you see why it's still appearing as not signed?

**Comment by mrpau-richard** [comment]
@aronasorman The reason why the installer is not signed, I designed it to sign an installer intentionally when it's needed. Because when we sign it every time there's a build, it will extend the build time. and we don't need that sign Windows installer unless it's a release build.

@radinamatic  to build a signed Windows installer just follow my comment [here.](https://github.com/learningequality/kolibri/pull/3702#issuecomment-392999808)

**Comment by indirectlylit** [comment]
address #2450

@aronasorman would you mind testing this when you have a chance?

**Review by aronasorman — APPROVED** [review]
Signing works

**Pull request merged**